database/dbmodel.py

```diff
@@ -448,24 +452,127 @@ class Reservation(db.Model):
 
     @staticmethod
     def get_reservation(reservation_id):
-        return Reservation.query.filter(Reservation.ID == reservation_id).first()
+        reservation = Reservation.query.filter(Reservation.ID == reservation_id).first()
+
+        if reservation:
+            return reservation
+        else:
+            raise ValueError('Reservation of ID "{}" does not exist'.format(str(reservation_id)))
+
+    @staticmethod
+    def get_reservations(start_date=date(2019, 1, 1), end_date=date(2099, 12, 31), show_cancelled=False):
+        reservation_list = []
+
+        if show_cancelled is True:
+            query = Reservation.query.filter(
+                Reservation.StartDate > start_date,
+                Reservation.EndDate < end_date
+            ).with_entities(Reservation.ID).all()
+        else:
+            query = Reservation.query.filter(
+                Reservation.Cancelled != True,
+                Reservation.StartDate > start_date,
+                Reservation.EndDate < end_date
+            ).with_entities(Reservation.ID).all()
+
+        print(query)
+        for reservation_id in query:
+            reservation = Reservation.get_reservation(reservation_id[0])
+            reservation_list.append(reservation)
+
+        return reservation_list
 
     def cancel(self):
         if self.Cancelled:
-            print("Reservation " + self.ID + " is already cancelled")
+            print("Reservation " + str(self.ID) + " is already cancelled")
             raise AttributeError
 
         self.Cancelled = True
         db.session.commit()
 
-    def set_machine_count(self, machine_count):
-        if machine_count <= self.MachineCount:
-            self.MachineCount = machine_count
-            db.session.commit()
+    def get_series(self, start_date=date.now(), end_date=date(2099, 12, 31), series_type='series'):
+        # series is defined by the same pool, same user and same weekday
```

Comment:
Nie byłabym za tym, że tylko weekday - pora również jest istotna (np. wtorek o 8:00 vs inny wtorek o 11:00 to wciąż seria)?